homeassistant/components/litterrobot/binary_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Support for Litter-Robot binary sensors."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic
+
+from pylitterbot import LitterRobot, Robot
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .entity import LitterRobotEntity, _RobotT
+from .hub import LitterRobotHub
+
+
+@dataclass
+class RequiredKeysMixin(Generic[_RobotT]):
+    """A class that describes robot binary sensor entity required keys."""
+
+    is_on_fn: Callable[[_RobotT], bool]
+
+
+@dataclass
+class RobotBinarySensorEntityDescription(
+    BinarySensorEntityDescription, RequiredKeysMixin[_RobotT]
+):
+    """A class that describes robot binary sensor entities."""
+
+
+class LitterRobotBinarySensorEntity(LitterRobotEntity[_RobotT], BinarySensorEntity):
+    """Litter-Robot binary sensor entity."""
+
+    entity_description: RobotBinarySensorEntityDescription[_RobotT]
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state."""
+        return self.entity_description.is_on_fn(self.robot)
+
+
+BINARY_SENSOR_MAP: dict[type[Robot], tuple[RobotBinarySensorEntityDescription, ...]] = {
+    LitterRobot: (
+        RobotBinarySensorEntityDescription[LitterRobot](
+            key="is_sleeping",
+            name="Is sleeping",
+            icon="mdi:sleep",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            entity_registry_enabled_default=False,
+            is_on_fn=lambda robot: robot.is_sleeping,
+        ),
+        RobotBinarySensorEntityDescription[LitterRobot](
+            key="sleep_mode_enabled",
+            name="Sleep mode enabled",
+            icon="mdi:sleep",
+            entity_category=EntityCategory.DIAGNOSTIC,
+            entity_registry_enabled_default=False,
+            is_on_fn=lambda robot: robot.sleep_mode_enabled,
+        ),
+    ),
+    Robot: (
+        RobotBinarySensorEntityDescription[Robot](
+            key="power_status",
+            name="Power status",
```

Comment:
Is this plugged in? Can we make this Charging?

Reply:
I'm not sure charging is the right concept for this. The unit is primarily mains powered and doesn't include a backup battery by default (except, I think, on the feeder robot). And nothing in the "API" indicates whether a battery is actually installed. So most of the time it's simply just "AC" in the data to indicate it's powered.

Reply:
Why does it even exist if it has no battery 😬

Reply:
It's powered on or it's unavailable ?

Reply:
Clarification, a battery isn't included by default, but you can buy one and install it so that it continues to function during power outages. I've installed one in all of my units. So then it comes back as "AC", "DC" or "NC" (not connected).